content/en-US/api/URLPattern.md

```diff
@@ -0,0 +1,515 @@
+---
+recipe: api-interface
+title: 'URLPattern'
+mdn_url: /en-US/docs/Web/API/URLPattern
+specifications: https://github.com/WICG/urlpattern/blob/main/explainer.md
+browser_compatibility: api.URLPattern
+---
+
+// For newer or recently updated specs, the preferred heading anchor is one
+// beginning with 'dom'.
+
+// Please do the following:
+// 1. Replace all [[tokens]] with the specified information. Be sure to remove
+// the brackets.
+// 2. Answer all questions in comments.
+// 3. Remove all comments.
+
+**When this feature ships, the content below will live on MDN under
+[developer.mozilla.org/en-US/docs/Web/API](https://developer.mozilla.org/en-US/docs/Web/API).**
+
+// Example interface page at: https://developer.mozilla.org/en-US/docs/Web/API/ContentIndex
+
+## Description
+
+// Complete the sentence below with a basic description of the API.
+// You may add more sentences if needed.
+
+The `URLPattern` interface provides a web platform primitive for matching
+URLs.  The pattern syntax is adopted from the popular [path-to-regexp][]
```

Comment:
It can be used for matching URLs in service workers, but no its not only for that.  You can use it for routing within a single page app in a document, for example.  It works in all window and worker javascript contexts.